src/microbit/__model/microbit_model.py

```diff
@@ -25,9 +28,21 @@ def running_time(self):
         print(f"time. time: {time.time()}")
         return time.time() - self.__start_time
 
+    def temperature(self):
+        return self.__temperature
+
+    def __set_temperature(self, temperature):
+        if (
+            temperature < CONSTANTS.MIN_TEMPERATURE
+            or temperature > CONSTANTS.MAX_TEMPERATURE
+        ):
+            raise ValueError(CONSTANTS.INVALID_TEMPERATURE_ERR)
+        else:
+            self.__temperature = temperature
+
     def update_state(self, new_state):

```

Comment:
should we split this part up into smaller functions? Like a function for temperature, function for accelerometer, etc because this could get quite big

Reply:
fixed!